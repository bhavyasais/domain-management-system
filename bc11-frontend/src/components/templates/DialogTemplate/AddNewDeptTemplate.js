import React, { useState } from "react";
import TextField from "@material-ui/core/TextField";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";
import ButtonAtom from "../../atoms/Button/ButtonAtom";
import { Typography } from "@material-ui/core";

export default function AddNewDeptTemplate(props) {

    const handleAddClick = () => {
        props.onAddClick(department);
        props.handleClose();
    }

    const [department,setDepartment]= useState('');
    const handleDepartmentChange = (e) => {
        setDepartment(e.target.value);
    }


  return (
    <div>
      <Dialog
        open={props.open}
        onClose={props.handleClose}
        aria-labelledby="form-dialog-title"
      >
        <DialogTitle id="form-dialog-title"><Typography variant="h6">Add New Department</Typography></DialogTitle>
        <DialogContent>
          <form>
            <div>
              <TextField
                id="outlined-basic"
                label="Department"
                variant="outlined"
                onChange={handleDepartmentChange}
              />
            </div>
          </form>
        </DialogContent>
        <DialogActions>
        <ButtonAtom
          
          variant="outlined"
          value="Close"
          textVariant="caption"
          onClick={props.handleClose}
        />
        <ButtonAtom
            variant="contained"
            value="Add"
            textVariant="caption"
            onClick={handleAddClick}
          />
        
           
        </DialogActions>
      </Dialog>
    </div>
  );
}
