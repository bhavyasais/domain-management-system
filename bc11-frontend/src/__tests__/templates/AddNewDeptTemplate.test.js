import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";
import { act } from "react-dom/test-utils";
import { TextField } from "@material-ui/core";
import AddNewDeptTemplate from "../../components/templates/DialogTemplate/AddNewDeptTemplate";

describe("<AddNewTabTemplate/>", () => {
  let mounted;
  const handleClose = jest.fn();
  const onAddClick = jest.fn();

  beforeEach(() => {
    act(() => {
      mounted = mount(
        <AddNewDeptTemplate
          handleClose={handleClose}
          onAddClick={onAddClick}
          open={true}
        />
      );
    });
  });

  it("Renders correctly", () => {
    let wrapper;
    wrapper = renderer.create(
      <AddNewDeptTemplate
        handleClose={() => {}}
        onConAddClicklick={() => {}}
        open={false}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('handles the change', () => {
    const wrapper = mount(
      <AddNewDeptTemplate onClick={onAddClick} handleClose={handleClose} onAddClick={onAddClick} open={true} onChange={onAddClick}/>,
    );
    wrapper.find('ButtonAtom').at(1).props().onClick();
    wrapper.find(TextField).at(0).props().onChange({target:{value:"abc"}});
  });
});
