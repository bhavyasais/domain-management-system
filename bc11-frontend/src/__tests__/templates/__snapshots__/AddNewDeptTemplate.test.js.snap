// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddNewTabTemplate/> Renders correctly 1`] = `<div />`;
